tools/buildmgr/cbuildgen/src/CMakeListsGenerator.cpp

```diff
@@ -450,6 +503,21 @@ bool CMakeListsGenerator::GenBuildCMakeLists(void) {
           cmakelists << "    set_source_files_properties(${SRC} PROPERTIES COMPILE_FLAGS \"${FILE_FLAGS}\")" << EOL;
           cmakelists << "  endif()" << EOL;
         }
+        if (specific_options) {
+          cmakelists << "  foreach(OPTION OPTIMIZE DEBUG WARNINGS)" << EOL;
+          cmakelists << "    if(DEFINED ${OPTION}_${S})" << EOL;
+          cmakelists << "      set(${OPTION}_LOCAL \"${${OPTION}_${S}}\")" << EOL;
+          cmakelists << "    else()" << EOL;
+          cmakelists << "      set(${OPTION}_LOCAL \"${${OPTION}}\")" << EOL;
+          cmakelists << "    endif()" << EOL;
+          cmakelists << "  endforeach()" << EOL;
+          cmakelists << "  get_source_file_property(FILE_FLAGS ${SRC} COMPILE_FLAGS)" << EOL;
+          cmakelists << "  if(FILE_FLAGS STREQUAL \"NOTFOUND\")" << EOL;
+          cmakelists << "    set(FILE_FLAGS)" << EOL;
+          cmakelists << "  endif()" << EOL;
+          cmakelists << "  cbuild_get_options_flags(" << lang << " \"${OPTIMIZE_LOCAL}\" \"${DEBUG_LOCAL}\" \"${WARNINGS_LOCAL}\" FILE_FLAGS)" << EOL;
```

Comment:
Maybe instead of directly using `FILE_FLAGS` here, you should store it in a temporary variable and then append it to `FILE_FLAGS` to avoid overwriting occasional previous values (e.g. from defines)?

Reply:
I was assuming the `cbuild_get_options_flags` call in line 518 was replacing the values of `FILE_FLAGS`, but it is actually prepending values into `FILE_FLAGS`. It should be ok, thanks for having checked it.